packages-exp/auth-exp/src/core/util/assert.ts

```diff
@@ -15,28 +15,72 @@
  * limitations under the License.
  */
 
+import { SDK_VERSION } from '@firebase/app-exp';
 import { AUTH_ERROR_FACTORY, AuthErrorCode } from '../errors';
+import { logError } from './log';
 
-export function assert<T>(
-  expression: T | null | undefined,
-  appName: string
-): T {
-  if (!expression) {
-    throw AUTH_ERROR_FACTORY.create(AuthErrorCode.INTERNAL_ERROR, { appName });
-  }
+/**
+ * Unconditionally fails, throwing a developer facing INTERNAL_ERROR
+ *
+ * @param appName App name for tagging the error
+ * @throws FirebaseError
+ */
+export function fail(appName: string): never {
+  throw AUTH_ERROR_FACTORY.create(AuthErrorCode.INTERNAL_ERROR, { appName });
+}
 
-  return expression;
+/**
+ * Verifies the given condition and fails if false, throwing a developer facing error
+ *
+ * @param assertion
+ * @param appName
+ */
+export function assert(assertion: boolean, appName: string): asserts assertion {
```

Comment:
No need to constrain `assertion` to boolean. This check is good for truthy values as well